include/interfaces/non_conservative/navier_stokes.h

```diff
@@ -286,6 +307,19 @@ declare_parameters (ParameterHandler &prm)
 {
   NonConservativeInterface<dim,dim,dim+1, NavierStokes<dim> >::declare_parameters(prm);
   this->add_parameter(prm, &rho,         "rho [kg m^3]",  "1.0", Patterns::Double(0.0));
+  
+  this->add_parameter(  prm,  &circular_obstacle,
+                        "Circular obstacle",
+                        "false",
+                        Patterns::Bool(),
+                        "Refine the obstable in order to have a circle.\n");
+
+  this->add_parameter(prm, &obstacle_boundary_id,
+                      "Obstacle boundary id",  "99", Patterns::Integer(0),
+                      "Refine the obstacle having this boundary id.\n");
```

Comment:
ok se è comodo a te. Per me in questo modo si limita l'utente ad avere mesh con un ostacolo solo con un manifold circolare. Per non togliere generalità lascerei all'utente di cambiare la `postprocess_newly_created_triangulation`. ma è solo la mia impressione a pelle :)